### PR TITLE
Add hot-reloading support to graphs.

### PR DESCRIPTION
**Pull request by maikelvl**
> If you have any ideas, please open a Pull Request

Here you go, I've added hot-reloading capabilities to your awesome mermaid plugin. 😄

It is inspired by https://github.com/ulivz/vuepress-plugin-flowchart

Hope you like it.

### CODE REVIEW

File: src/markdownItPlugin.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  }
+
+  // Handle <mermaid> blocks
+  options = options || {}
+  const openMarker = options.openMarker || '<mermaid'  // Not ending with '>' to allow attributes
+  const openChar = openMarker.charCodeAt(0)
+  const closeMarker = options.closeMarker || '</mermaid>'
+  const closeChar = closeMarker.charCodeAt(0)
+
+  function render (tokens, idx, options, env, self) {
```
Comment (eFrane) [inline src/markdownItPlugin.js]: Can you please add a short doc comment to the function?

File: src/markdownItPlugin.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    const token = tokens[idx]
+    const key = `mermaid_${hash(idx)}`
+    const { content } = token
+    md.$dataBlock[key] = content
+    return `<Mermaid id="${key}" :graph="$dataBlock.${key}"></Mermaid>`
+  }
+
+  function uml (state, startLine, endLine, silent) {
```
Comment (eFrane) [inline src/markdownItPlugin.js]: Same here, doc comment please. Also, I would prefer having this renamed to mermaidRender or something to be a little more explicit.

**Comment by maikelvl** [comment]
> I like what you did there. However, this does not seem to work when doing a static build of a vuepress site (try `yarn vuepress build docs` in this repository.

The docs build is working on my machine using node v10.19.0. How does it not work? Did you get an error?

File: src/markdownItPlugin.js (hunk 1 above)
Comment (maikelvl) [reply src/markdownItPlugin.js]: Good point. I've added comments. 😄

File: src/markdownItPlugin.js (hunk 2 above)
Comment (maikelvl) [reply src/markdownItPlugin.js]: Good point. I've added comments. 😄

**Comment by eFrane** [comment]
I get errors on my machine (node v12.10.0), plus the checks aka the Netlify build of the docs site fail.

**Comment by eFrane** [comment]
Code-wise this looks good now, I however don't feel good merging it with at least partially breaking dist builds.

I have attached the error output from running `vuepress build` on my machine.

```
Rendering page: /error Error rendering /: false
undefined
ReferenceError: window is not defined
    at Object.<anonymous> (node_modules/mermaid/dist/mermaid.js:10:1)
    at __webpack_require__ (webpack/bootstrap:25:0)
    at Module.<anonymous> (server-bundle.js:103836:15)
    at __webpack_require__ (webpack/bootstrap:25:0)
    at Object.<anonymous> (server-bundle.js:102690:18)
    at __webpack_require__ (webpack/bootstrap:25:0)
    at server-bundle.js:118:18
    at Object.<anonymous> (server-bundle.js:121:10)
    at evaluateModule (.../vuepress-plugin-mermaidjs/node_modules/vue-server-renderer/build.dev.js:9303:21)
    at .../vuepress-plugin-mermaidjs/node_modules/vue-server-renderer/build.dev.js:9361:18
    at new Promise (<anonymous>)
    at .../vuepress-plugin-mermaidjs/node_modules/vue-server-renderer/build.dev.js:9353:14
    at Object.renderToString (.../vuepress-plugin-mermaidjs/node_modules/vue-server-renderer/build.dev.js:9529:9)
    at Build.renderPage (.../vuepress-plugin-mermaidjs/node_modules/@vuepress/core/lib/node/build/index.js:156:34)
    at Build.render (.../vuepress-plugin-mermaidjs/node_modules/@vuepress/core/lib/node/build/index.js:94:33)
    at async App.build (.../vuepress-plugin-mermaidjs/node_modules/@vuepress/core/lib/node/App.js:494:5)
error Command failed with exit code 1.
```

**Comment by maikelvl** [comment]
I've moved the mermaid library import back to the 'mounted' method and now the checks passed.

**Comment by eFrane** [comment]
Yes, it looks good now on my end too. Thank you for the contribution!

**Pull request merged**